src/Model/Behavior/FootprintBehavior.php

```diff
@@ -76,13 +76,16 @@ public function initialize(array $config)
     public function beforeFind(Event $event, Query $query, ArrayObject $options)
     {
         $eventName = $event->name();
-        $config = (array)$this->config('events.' . $eventName);
+        $config = Hash::normalize((array)$this->config('events.' . $eventName));
 
         foreach (array_keys($config) as $field) {
             $path = $this->config('propertiesMap.' . $field);
-            $query->where([
-                $field => current(Hash::extract((array)$options, $path))
-            ]);
+            $value = current(Hash::extract((array)$options, $path));
```

Comment:
Using `Hash::get()` instead of `Hash::extract()` would avoid having to use `current()`. Same below.